frontend/azlinux/common.go

```diff
@@ -0,0 +1,58 @@
+package azlinux
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/Azure/dalec"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+)
+
+func installWithReqs(target string, w worker, deps map[string]dalec.PackageConstraints, installOpts ...installOpt) installFunc {
+	// depsOnly is a simple dalec spec that only includes build dependencies and their constraints
+	depsOnly := dalec.Spec{
+		Name:        "azlinux-build-dependencies",
+		Description: "Provides build dependencies for mariner2 and azlinux3",
+		Version:     "1.0",
+		License:     "Apache 2.0",
+		Revision:    "1",
+		Dependencies: &dalec.PackageDependencies{
+			Runtime: deps,
+		},
+	}
+
+	return func(ctx context.Context, client gwclient.Client, sOpt dalec.SourceOpts) (llb.RunOption, error) {
+		pg := dalec.ProgressGroup("Building container for build dependencies")
+
+		// create an RPM with just the build dependencies, using our same base worker
+		rpmDir, err := specToRpmLLB(ctx, w, client, &depsOnly, sOpt, target, pg)
+		if err != nil {
+			return nil, err
+		}
+
+		// read the built RPMs (there should be a single one)
+		rpms, err := readRPMs(ctx, client, rpmDir)
```

Comment:
This is failry expensive since it un-lazies the build.
Do we need to read the rpm list vs installing all returned RPM's?

Reply:
This is a good point. I think you're right, in this case we can actually just install all returned RPMs